metadata.go

```diff
@@ -0,0 +1,147 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sdk
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/conduitio/conduit-connector-protocol/cpluginv1"
+)
+
+const (
+	// MetadataOpenCDCVersion is a Record.Metadata key for the version of the
+	// OpenCDC format (e.g. "v1"). This field exists to ensure the OpenCDC
+	// format version can be easily identified in case the record gets marshaled
+	// into a different untyped format (e.g. JSON).
+	MetadataOpenCDCVersion = cpluginv1.MetadataOpenCDCVersion
+
+	// MetadataCreatedAt is a Record.Metadata key for the time when the record
+	// was created in the 3rd party system. The expected format is a unix
+	// timestamp in nanoseconds.
+	MetadataCreatedAt = cpluginv1.MetadataCreatedAt
+	// MetadataReadAt is a Record.Metadata key for the time when the record was
+	// read from the 3rd party system. The expected format is a unix timestamp
+	// in nanoseconds.
+	MetadataReadAt = cpluginv1.MetadataReadAt
+
+	// MetadataConduitPluginName is a Record.Metadata key for the name of the
+	// plugin that created this record.
+	MetadataConduitPluginName = cpluginv1.MetadataConduitPluginName
+	// MetadataConduitPluginVersion is a Record.Metadata key for the version of
+	// the plugin that created this record.
+	MetadataConduitPluginVersion = cpluginv1.MetadataConduitPluginVersion
+)
+
+// MetadataUtil provides utility methods for reading and writing metadata.
+type MetadataUtil struct{}
```

Comment:
Would it make sense to have a metadata type which would have map[string]string as the undelying type, and which we would use in Record? That way, we could have the method below directly on metadata, and it would be easier to call them. E.g.
```
type metadata map[string]string

func (m metadata) SetFoo() {
	m["foo"] = "bar"
}

type Record struct {
    m metadata
    ...
}
``` 
I'm not quite sure how would it affect serialization though.

Reply:
Good idea, I tried it out and I think it looks nice 👍 lmk what you think about about the last commit

Reply:
Looks good!